packages/sync/src/createClient.ts

```diff
@@ -1,23 +1,31 @@
-import { InMemoryCache } from "apollo-cache-inmemory";
+import { InMemoryCache, NormalizedCacheObject } from "apollo-cache-inmemory";
 import { persistCache } from "apollo-cache-persist";
-import { ApolloClient } from "apollo-client";
+import { ApolloClient, ApolloClientOptions } from "apollo-client";
 import { DataSyncConfig } from "./config/DataSyncConfig";
 import { SyncConfig } from "./config/DefaultConfig";
 import { defaultLinkBuilder as buildLink} from "./links/LinksBuilder";
 import { PersistedData, PersistentStore } from "./PersistentStore";
+import { SyncOfflineMutation } from "./offline/SyncOfflineMutation";
 
 /**
  * Factory for creating Apollo Client
  *
- * @param options options object used to build client
+ * @param userConfig options object used to build client
  */
 export const createClient = async (userConfig?: DataSyncConfig) => {
 
   const clientConfig = extractConfig(userConfig);
-  const cache = await buildStorage(clientConfig);
-  const link = buildLink(clientConfig);
-
-  return new ApolloClient({ link , cache });
+  const { cache, storage } = await buildStorage(clientConfig);
+  const link = buildLink(clientConfig, storage);
+  const options: ApolloClientOptions<NormalizedCacheObject> = {
+    link,
+    cache
+  };
+  const apolloClient = new ApolloClient<NormalizedCacheObject>(options);
+  const syncOfflineMutations = new SyncOfflineMutation(apolloClient, storage, clientConfig.mutationsQueueName);
+  await syncOfflineMutations.init();
+  await syncOfflineMutations.sync();
```

Comment:
I know that I have done that initially. But locking rendering until entire network offline queue will be process can seriously slow down application loading.  Users may see black screen  for a while. We can do that asynchronously in the background